conference.js

```diff
@@ -700,22 +719,42 @@ export default {
     /**
      * Simulates toolbar button click for video mute. Used by shortcuts and API.
      * @param mute true for mute and false for unmute.
-     */
-    muteVideo(mute) {
-        muteLocalVideo(mute);
+     * @param {boolean} [noUI] when set to true will not display any error
+     * dialogs in case of media permissions error.
+     */
+    muteVideo(mute, noUI = false) {
+        if (!localVideo && this.videoMuted && !mute) {
+            // Try to create local video if there wasn't any.
+            // This handles the case when user joined with no video
+            // (dismissed screen sharing screen or in audio only mode), but
+            // decided to add it later on by clicking on muted video icon or
+            // turning off the audio only mode.
+            //
+            // FIXME when local track creation is moved to react/redux
+            // it should take care of the use case described above
+            createLocalTracks({ devices: ['video'] }, false)
+                .then(([videoTrack]) => {
+                    this.useVideoStream(videoTrack);
+                })
+                .catch(error => {
+                    // FIXME should send some feedback to the API on error ?
+                    if (!noUI) {
+                        APP.UI.showDeviceErrorDialog(null, error);
+                    }
```

Comment:
Would it be preferable to make the argument name showUI to avoid a double negative?